### ISSUE TITLE
Prisma seems to be looking in the wrong location for rhel-openssl-1.0.x

### ISSUE THREAD

**Issue by RoryKelly** [prisma/prisma#15638]
### Bug description

in  a monorepo project, I am building for AWS.  I cannot use the `prisma-serverless-webpack` plugin as it does not work with monorepos.

When calling prisma I get the following error. 

```
Query engine library for current platform \rhel-openssl-1.0.x\ could not be found.
You incorrectly pinned it to rhel-openssl-1.0.x

This probably happens because you built Prisma Client on a different platform.
(Prisma Client looked in \/var/task/src/libquery_engine-rhel-openssl-1.0.x.so.node\)

Searched Locations:

  /.prisma/client
  /home/runner/work/8bars/8bars/backend-v2/node_modules/@prisma/client
  /var/task
  /var/task/src
  /var/task/src
  /tmp/prisma-engines
  /var/task/src

You already added the platforms \debian-openssl-1.1.x\ \rhel-openssl-1.0.x\ to the \generator\ block
in the \schema.prisma\ file as described in https://pris.ly/d/client-generator
but something went wrong. That's suboptimal.

Please create an issue at https://github.com/prisma/prisma/issues/new
```

The correct file is included in the project but at the location 

```
 /home/runner/work/8bars/8bars/backend-v2/node_modules/.prisma/client
```

See:

<img width="1103" alt="Screenshot 2022-10-02 at 20 21 58" src="https://user-images.githubusercontent.com/1162214/193472324-c8011ecf-176b-4db8-b36f-92ce8edaa1c7.png">

This is not in the list of the Searched Locations.

### How to reproduce

Build a prisma client for AWS without using prisma-serverless-webpack. 

### Expected behavior

The aws Lambda runs correctly.

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```
4.4.0
```

**Comment by janpio** [comment]
Hey @RoryKelly , can you share a reproduction of your project setup so we can reproduce this? There are many underlying problems that can causethis, so we would like to be able to debug and reproduce ourselves optimally. Thanks!

**Comment by millsp** [comment]
Hey, we wanted to give you an update about this. We have around 10 issues that we have identified to be in the "engine not found" category. Looking at these, we learned that there are four main categories responsible for these errors:

* Your client was generated with `native` (default) on one platform, but used on a different platform at runtime.
* You pinned `binaryTargets` incorrectly, and the correct engine was therefore not included at runtime.
* A bundler/framework changed the Prisma Client but did not copy the necessary engine for the runtime.
* Deployment/tooling did not copy the engine correctly so the correct engine was not included at runtime.

These cases were previously not detected. On top of that, the error messages were confusing and were neither helpful for you or us. With that, it is likely that many of the cases above overlapped each other. So while we cannot fix a missing engine, we can detect how and why it happened. Thanks to that, we now display more useful errors to help you out of trouble. And in some cases, we will ask some input from you to learn more about your project so we can further improve.

If something is still off for you, feel free to create a new issue.
